update/sync/coordinator.go

```diff
@@ -0,0 +1,481 @@
+	}
+	if check.IfNil(args.HeaderValidator) {
+		return nil, process.ErrNilHeaderValidator
+	}
+
+	ss := &syncState{
+		hasher:                 args.Hasher,
+		marshalizer:            args.Marshalizer,
+		shardCoordinator:       args.ShardCoordinator,
+		trieSyncers:            args.TrieSyncers,
+		epochHandler:           args.EpochHandler,
+		requestHandler:         args.RequestHandler,
+		headerValidator:        args.HeaderValidator,
+		tries:                  make(map[string]data.Trie),
+		syncingEpoch:           0,
+		activeAccountsAdapters: args.AccountHandlers,
+	}
+
+	ss.miniBlocks = pendingMiniBlocks{
+		mutPendingMb:  sync.Mutex{},
+		mapMiniBlocks: make(map[string]*block.MiniBlock),
+		mapHashes:     make(map[string]struct{}),
+		pool:          args.DataPools.MiniBlocks(),
+		storage:       args.Storages.GetStorer(dataRetriever.MiniBlockUnit),
+		chReceivedAll: make(chan bool),
+	}
+	if check.IfNil(ss.miniBlocks.storage) {
+		return nil, update.ErrNilMiniBlocksStorage
+	}
+
+	ss.transactions = pendingTransactions{
+		mutPendingTx:    sync.Mutex{},
+		mapTransactions: make(map[string]data.TransactionHandler),
+		mapHashes:       make(map[string]*block.MiniBlock),
+		chReceivedAll:   make(chan bool),
+	}
+
+	ss.transactions.txPools = make(map[block.Type]dataRetriever.ShardedDataCacherNotifier)
+	ss.transactions.txPools[block.TxBlock] = args.DataPools.Transactions()
+	ss.transactions.txPools[block.SmartContractResultBlock] = args.DataPools.UnsignedTransactions()
+	ss.transactions.txPools[block.RewardsBlock] = args.DataPools.RewardTransactions()
+
+	ss.transactions.storage = make(map[block.Type]update.HistoryStorer)
+	ss.transactions.storage[block.TxBlock] = args.Storages.GetStorer(dataRetriever.TransactionUnit)
+	ss.transactions.storage[block.SmartContractResultBlock] = args.Storages.GetStorer(dataRetriever.UnsignedTransactionUnit)
+	ss.transactions.storage[block.RewardsBlock] = args.Storages.GetStorer(dataRetriever.RewardTransactionUnit)
+
+	for _, pool := range ss.transactions.txPools {
+		pool.RegisterHandler(ss.receivedTransaction)
+	}
+
+	for _, store := range ss.transactions.storage {
+		if check.IfNil(store) {
+			return nil, epochStart.ErrNilStorage
+		}
+	}
+
+	ss.transactions.chReceivedAll = make(chan bool)
+	ss.miniBlocks.chReceivedAll = make(chan bool)
+
+	ss.miniBlocks.pool.RegisterHandler(ss.receivedMiniBlock)
+
+	return ss, nil
+}
+
+// SyncAllState gets an epoch number and will sync the complete data for that epoch start metablock
+func (ss *syncState) SyncAllState(epoch uint32) error {
+	if epoch == ss.syncingEpoch {
+		return nil
+	}
+	ss.syncingEpoch = epoch
+
+	meta, err := ss.headers.SyncEpochStartMetaHeader(epoch, time.Minute)
+	if err != nil {
+		return err
+	}
+
+	ss.syncingEpoch = meta.GetEpoch()
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(meta.EpochStart.LastFinalizedHeaders) + 1)
+
+	var errFound error
+	mutErr := sync.Mutex{}
+
+	go func() {
+		errMeta := ss.syncMeta(meta, &wg)
```

Comment:
`ss.syncMeta` seems to be missing after the last push

Reply:
done.